File: algo-method.com/tasks/1090/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const even = (n: number) => n % 2 === 0;

const main = () => {
  const [H, W, P, Q] = input().split(" ").map(Number);

  const ans = even(P + Q) ? "Black" : "White";

  console.log(ans);
};

main();

File: algo-method.com/tasks/1090/main.cpp
#include <bits/stdc++.h>
using namespace std;

bool even(int n) { return n % 2 == 0; }

int main() {
  int H, W, P, Q;
  cin >> H >> W >> P >> Q;

  string ans = even(P + Q) ? "Black" : "White";

  cout << ans << endl;
}
